ansible/playbooks/AdoptOpenJDK_AIX_Playbook/roles/X11/tasks/main.yml

```diff
@@ -1,46 +1,86 @@
 #########################################################################
 # Install X11 extensions                                                #
+# x11.base.lib installp download requiring an IBMid                     #
+# https://www.ibm.com/support/pages/fileset-information-x11baselib      #
 # x11.adt.ext installp download requiring an IBMid                      #
-# https://www-01.ibm.com/support/docview.wss?uid=isg1fileset-1198156818 #
+# https://www.ibm.com/support/pages/fileset-information-x11adtext       #
 # x11.vfb installp download requiring an IBMid                          #
-# http://www-01.ibm.com/support/docview.wss?uid=isg1fileset-2054486724  #
+# https://www.ibm.com/support/pages/fileset-information-x11vfb          #
 #########################################################################
 ---
-- name: Check if X11.adt.ext is installed
-  shell: lslpp -l X11.adt.ext >/dev/null 2>&1
-  register: does_X11_adt_ext_exist
-  ignore_errors: yes
+# Rather than run lslpp - use stat on the lpp directories
+# /usr/lpp/X11.base/X11.base.lib      X11.base.lib
+# /usr/lpp/X11/include/X11/extensions X11.adt.ext
+# /usr/lpp/X11.vfb                    X11.vfb
+- name: Verify X11 software
+  block:
+  - name: Verify X11 base library
+    stat:
+      path: /usr/lpp/X11.base/X11.base.lib
+    register: _base
+  - name: Verify X11 extensions requirements
+    stat:
+      path: /usr/lpp/X11/include/X11/extensions
+    register: _ext
+  - name: Verify X11 Frame Buffer Requirements
+    stat:
+      path: /usr/lpp/X11.vfb
+    register: _vfb
+  - name: Verify Readiness #1
+    set_fact:
+      _x11rdy: "{{ _base.stat.exists
+        and _ext.stat.exists
+        and _vfb.stat.exists }}"
   tags: x11
 
-- name: Check if X11.vfb is installed
-  shell: lslpp -l X11.vfb >/dev/null 2>&1
-  register: does_x11_exist
-  ignore_errors: yes
-  tags: x11
+- name: Install from Vendor Files
+  block:
+    - name: Unpack Vendor Files
+      unarchive:
+        src: /Vendor_Files/aix/OpenGL_X11.tar
+        dest: /tmp
+        remote_src: no
+      ignore_errors: yes
+      register: _vendor_copied
 
-- name: Transfer and extract X11
-  unarchive:
-    src: /Vendor_Files/aix/OpenGL_X11.tar
-    dest: /tmp
-    remote_src: no
-  when: does_x11_exist.rc != 0 or does_X11_adt_ext_exist.rc != 0
-  tags: x11
+# X11.base.lib is not included in the so-called vendor files
+# So no attempt is made to install it.
+# Still reports if it is missing
+    - name: Install IBM X11 Extensions - installp
+      command:
+        installp -e /tmp/x11.log -aXYgd /tmp/x11 X11.adt.ext X11.vfb
```

Comment:
The single installp command will still only install X11.vfb and not X11.adt.ext

```
root@p9-aix1-ojdk06:[/tmp]installp -e /tmp/x11.log -aXYgd /tmp/x11 X11.adt.ext X11.vfb
+-----------------------------------------------------------------------------+
                    Pre-installation Verification...
+-----------------------------------------------------------------------------+
Verifying selections...done
Verifying requisites...done
Results...

FAILURES
--------
  Filesets listed in this section failed pre-installation verification
  and will not be installed.

  Missing Filesets
  ----------------
  The following filesets could not be found on the installation media.
  If you feel these filesets really are on the media, check for typographical
  errors in the name specified or, if installing from directory, check for
  discrepancies between the Table of Contents file (.toc) and the images that
  reside in the directory.

    X11.adt.ext                              

  << End of Failure Section >>
```
To install X11.adt.ext, the install directory will need to point to /tmp/x11/X11.adt which will require a separate installp command, unless installp has a flag where you can provide two install directories

Reply:
I suspect the table of contents in the TAR file are not current - so added steps to remove previous x11.log (if present) and also any /tmp/x11/.toc - so that installp will create a new one - consistent with the directory contents.